keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resolvers/ImageResolver.kt

```diff
@@ -85,7 +85,8 @@ class ImageResolver(
     val images = imageService.getLatestNamedImages(
       appVersion = artifactVersion.parseAppVersion(),
       account = account,
-      regions = regions
+      regions = regions,
+      baseOs = artifact.vmOptions.baseOs
     )
```

Comment:
This is pretty critical -- when we resolve the desired state of a cluster, make sure we're looking for an image with the right base OS.